### PR TITLE
Rewrite unnecessary KartMove ASM patches

### PR DESCRIPTION
**Pull request by GnomedDev**
I can't be bothered to decomp the rest as they are large so we still have to have some ASM patches.

### CODE REVIEW

File: payload/game/kart/KartMove.cc [hunk 1]
```diff
@@ -16,6 +15,19 @@ const Vec3 *KartMove::internalVelDir() const {
     return &m_internalVelDir;
 }
 
+void KartMove::init(u32 r4, u32 r5) {
+    REPLACED(init)(r4, r5);
+    m_hardSpeedLimit = m_hardSpeedLimit * g_speedModFactor;
+}
+
+void KartMove::setBulletSpeedLimit() {
+    m_hardSpeedLimit = 145 * g_speedModFactor;
```
Comment (stblr) [inline payload/game/kart/KartMove.cc:24]:
```suggestion
    m_hardSpeedLimit = 145.0f * g_speedModFactor;
```

File: payload/game/kart/KartMove.cc [hunk 2]
```diff
@@ -16,6 +15,19 @@ const Vec3 *KartMove::internalVelDir() const {
...
+}
+
+void KartMove::restoreBulletSpeedLimit() {
+    m_hardSpeedLimit = 120 * g_speedModFactor;
```
Comment (stblr) [inline payload/game/kart/KartMove.cc:28]:
```suggestion
    m_hardSpeedLimit = 120.0f * g_speedModFactor;
```

File: payload/game/kart/KartMove.hh [hunk 3]
```diff
@@ -13,15 +13,28 @@ class KartMove : public KartObjectProxy {
...
+    KartMove();
+    virtual ~KartMove();
+    virtual void dt(s32 type);
+    virtual void *createComponents();
+    virtual void *setTurnParams();
+    void REPLACED(init)(u32 r4, u32 r5);
+    REPLACE virtual void init(u32 r4, u32 r5);
+
+    REPLACE void setBulletSpeedLimit();
+    REPLACE void restoreBulletSpeedLimit();
```
Comment (stblr) [inline payload/game/kart/KartMove.hh:26]: These names are confusing.

File: payload/game/kart/KartMove.cc [hunk 4]
```diff
@@ -36,4 +48,9 @@ bool KartMove::activateTcLightning() {
     return REPLACED(activateTcLightning)();
 }
 
+void KartMove::activateBullet(u32 r4) {
+    REPLACED(activateBullet)(r4);
+    setBulletSpeedLimit();
+}
```
Comment (stblr) [inline payload/game/kart/KartMove.cc:54]: Looking at 8059b7b8, we can maybe drop this.

**Review by stblr — CHANGES_REQUESTED** [review]

File: payload/game/kart/KartMove.hh (hunk 3 above)
Comment (GnomedDev) [reply payload/game/kart/KartMove.hh:26]: Would `restoreNormalSpeedLimit` be cleaner? Or what do you suggest otherwise?

Comment (stblr) [reply payload/game/kart/KartMove.hh:26]: Something that makes it clear that these functions are a pair like `applyBulletSpeedLimit`/`unapplyBulletSpeedLimit`.

**Review by stblr — APPROVED** [review]

**Pull request merged**